web/packages/url-utils/src/index.ts

```diff
@@ -0,0 +1,126 @@
+import { sha256 } from 'js-sha256';
+
+const AVATAR_BASE_URL = 'https://www.gravatar.com/avatar/';
+const PROFILE_BASE_URL = 'https://www.gravatar.com/';
+
+// Gravatar profile formats
+export enum GravatarFormat {
+	HTML = '',
+	JSON = 'json',
+	XML = 'xml',
+	PHP = 'php',
+	VCF = 'vcf',
+	QR = 'qr',
+}
+
+// Gravatar default images
+export enum GravatarDefault {
+	STATUS_404 = '404',
+	MYSTERY_PERSON = 'mp',
+	IDENTICON = 'identicon',
+	MONSTERID = 'monsterid',
+	WAVATAR = 'wavatar',
+	RETRO = 'retro',
+	ROBOHASH = 'robohash',
+	TRANSPARENT_PNG = 'blank',
+}
+
+// Gravatar ratings
+export enum GravatarRating {
+	GENERAL = 'g',
+	PARENTAL_GUIDANCE = 'pg',
+	RESTRICTED = 'r',
+	X = 'x',
+}
+
+interface GravatarAvatarOptions {
+	default?: GravatarDefault | string;
+	size?: number;
+	rating?: GravatarRating;
+	forceDefault?: boolean;
+}
+
+// Function to get the SHA256 hash of the email
+function getHash( email: string ): string {
+	if ( typeof email !== 'string' || email.trim() === '' ) {
+		throw new Error( 'Valid email is required' );
+	}
+	return sha256( email.trim().toLowerCase() );
+}
+
+// Function to validate the options
+function validateOptions( options: GravatarAvatarOptions ): void {
+	if ( options.size !== undefined ) {
+		if ( typeof options.size !== 'number' || options.size <= 0 ) {
+			throw new Error( "Invalid value for 'size'. It must be a positive number." );
```

Comment:
We often use `throw new Error('...')`, which halts the execution of the site or app when an error occurs. This is typically reserved for signaling critical issues that should stop the rest of the code from running.

For instance, if a user provides an incorrect option when rendering [a Gravatar component](https://github.com/Automattic/gravatar/pull/96/files#diff-3245f04c7d62226a16674b93862491730233df661b9684dae02f0e2e86fe61b1R121-R130), the entire site will stop functioning.

For non-critical issues, I recommend using `console.error('...')` instead, which allows the app to continue running while logging the error.